lib/inc/internal/nodes/config_node_simple_value.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "abstract_config_node.hpp"
+#include <internal/values/abstract_config_value.hpp>
+
+namespace hocon {
+
+    class config_node_simple_value : public abstract_config_node {
+    public:
+        config_node_simple_value(shared_token value);
+
+        shared_token get_token() const;
+        std::unique_ptr<abstract_config_value> const& get_value() const;
```

Comment:
You should be able to return an abstract reference. We probably should look more closely at how config_values are being used, it would make sense to use shared_ptr everywhere if they're expected to be owned by multiple objects.